rabbitmq/resource_queue.go

```diff
@@ -132,12 +132,33 @@ func ReadQueue(d *schema.ResourceData, meta interface{}) error {
 	e := make(map[string]interface{})
 	e["durable"] = queueSettings.Durable
 	e["auto_delete"] = queueSettings.AutoDelete
-	e["arguments"] = queueSettings.Arguments
-	queue[0] = e
 
-	d.Set("settings", queue)
+	// NOTE the addition of `arguments_json` never actually worked
+	// https://github.com/terraform-providers/terraform-provider-rabbitmq/pull/6
+	// d.Set("settings", queue) failed silently as a result of not checking the error.
+	// Additionally there would be a permadiff between the configuration and schema
+	// as the queue previously always set `arguments` on read. Issue was filed
+	// https://github.com/terraform-providers/terraform-provider-rabbitmq/issues/16
+	// In order to suppress the permadiff and error at the same time, if the
+	// data originated from `arguments_json` store the data in that key as a
+	// json string. We can feel confident in the data's origin because the schema
+	// was setup to conflict if both `arguments` and `arguments_json` were present
+	// This fix will not account for the case where `arguments` was used to supply
+	// string arguments to a queue, then the queue had arguments added outside of terraform,
+	// with values others than string, then d.Set("settings", queue) would fail.
```

Comment:
I'm not sure we need the full bug history of this portion of the code here... could the history live in the commit message instead, and then limit the comment here to concisely describe the reason for the _current_ behavior?

```go
// The user may have used either arguments or arguments_json to populate this originally.
// We need to preserve that decision here so that a subsequent Terraform plan for the
// same configuration wouldn't produce an errant diff that moves the value from one
// to the other without changing any values.
// These two arguments are mutually exclusive due to ConflictsWith in the schema.
```